### PR TITLE
Content Models: supporting preprocess rules and enhancements and

### PR DESCRIPTION
**Pull request by ilaner**
## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/XSUP-25326?filter=-1

## Description
* Add PreProcessRule content item
* Handle correctly a case where `marketplace` field in `pack_metadata.json` is an empty list.
* Identify LayoutContainer by the data and not by name.

### CODE REVIEW

File: demisto_sdk/commands/content_graph/objects/pre_process_rule.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+from typing import Set
+
+from demisto_sdk.commands.content_graph.common import ContentType
+from demisto_sdk.commands.content_graph.objects.base_content import BaseContent
+
+
+class PreProcessRule(BaseContent, content_type=ContentType.PREPROCESS_RULE):
+    def metadata_fields(self) -> Set[str]:
+        raise NotImplementedError("PreprocessRule not included in metadata")
```
Comment (dantavori) [inline demisto_sdk/commands/content_graph/objects/pre_process_rule.py:9]: why not? also, where will it be caught?

**Review by dantavori — COMMENTED** [review]
please run a test-upload against content master

File: TestSuite/pack.py [hunk 2]
```diff
@@ -451,6 +451,8 @@ def create_layout(self, name, content: dict = None) -> JSONBased:
         return layout
 
     def create_layoutcontainer(self, name, content: dict = None) -> JSONBased:
```
Comment (dorschw) [inline TestSuite/pack.py:453]: `Optional[dict]`

**Review by dorschw — COMMENTED** [review]
Nice. Do we have unit tests?

File: demisto_sdk/commands/content_graph/objects/pre_process_rule.py (hunk 1 above)
Comment (ilaner) [reply demisto_sdk/commands/content_graph/objects/pre_process_rule.py]: The server expects `name` and `description`, so I will show those.
However, we still have an issue where our `metdata.json` is not the same `metadata.json` the Upload Flow produces which will be fixed by @yaakovpraisler

File: demisto_sdk/commands/content_graph/parsers/pack.py [hunk 3]
```diff
@@ -123,8 +126,8 @@ def __init__(self, metadata: Dict[str, Any]) -> None:
         self.vendor_id: Optional[str] = metadata.get("vendorId")
         self.vendor_name: Optional[str] = metadata.get("vendorName")
         self.preview_only: Optional[bool] = metadata.get("previewOnly")
-        self.marketplaces: List[MarketplaceVersions] = metadata.get(
-            "marketplaces", DEFAULT_MARKETPLACES
+        self.marketplaces: List[MarketplaceVersions] = (
+            metadata.get("marketplaces") or DEFAULT_MARKETPLACES
         )
```
[the remark this answers is not in the document]
Comment (ilaner) [reply demisto_sdk/commands/content_graph/parsers/pack.py:131]: Yes, we don't support empty marketplaces value.

**Review by dorschw — APPROVED** [review]

**Comment by ilaner** [comment]
triggered SDK nightly for it:
https://code.pan.run/xsoar/content/-/pipelines/5566084

**Pull request merged**